### PR TITLE
 Fix upper bounds added for DataFrames 0.11

### PR DESCRIPTION
**Pull request by nalimilan**
Same as PR https://github.com/JuliaLang/METADATA.jl/pull/12220 but for DataFrames. This requires adding a lot of Julia 0.6 upper bounds

```
BioMedQuery: added julia 0.6 upper bound to v0.1.0 since only oldest release did not depend on DataFrames
Bootstrap: added julia 0.6 upper bound to versions before 0.0.3 since only old releases did not depend on DataFrames
CSV: already has an upper bound on julia 0.5
DSGE: added julia 0.6 upper bound to versions before 0.1.5 since only old releases did not depend on DataFrames
DataStreams: recent versions do not depend on DataFrames
Diversity: added julia 0.6 upper bound to versions before 0.3.0 since only old releases did not depend on DataFrames
GLM: only old versions depended on DataFrames, so OK
GLMNet: only old versions depended on DataFrames, so OK
GeoStats: recent versions do not depend on DataFrames [add bounds to older releases anyway?]
Graphs: only the old fist version depended on DataFrames, so OK
JWAS: added julia 0.6 upper bound to versions before 0.0.4 since only old releases did not depend on DataFrames
JuliaFEM: added julia 0.6 upper bound to versions before 0.2.1 since only old releases did not depend on DataFrames
MLDataUtils: added julia 0.6 upper bound to versions before 0.2.1 since only old releases did not depend on DataFrames
MarketTechnicals: only old versions depended on DataFrames, so OK
MixedModels: already has an upper bound on julia 0.6
Mustache: only old versions depended on DataFrames, so OK
ODBC: recent versions do not depend on DataFrames [add bounds to older releases anyway?]
OpenGene: added julia 0.6 upper bound to v0.1.0 since only oldest release did not depend on DataFrames
OptiMimi: added julia 0.6 upper bound to v0.0.1 since only oldest release did not depend on DataFrames
PGFPlots: added julia 0.6 upper bound to versions before 1.4.3 since only old releases did not depend on DataFrames
PlotlyJS: added julia 0.6 upper bound to versions before 0.6.0 since only old releases did not depend on DataFrames
Quandl: added julia 0.6 upper bound to versions before 0.3.0 since only old releases did not depend on DataFrames
RLEVectors: recent versions do not depend on DataFrames [add bounds to older releases anyway?]
Rif: added julia 0.6 upper bound to v0.0.0 since only oldest release did not depend on DataFrames
SALSA: only old versions depended on DataFrames, so OK
SQLite: recent versions do not depend on DataFrames [add bounds to older releases anyway?]
ScikitLearn: added julia 0.6 upper bound to versions before 0.3.0 since only old releases did not depend on DataFrames
Stan: recent versions do not depend on DataFrames [add bounds to older releases anyway?]
SymPy: recent versions do not depend on DataFrames [add bounds to older releases anyway?]
TermWin: added julia 0.6 upper bound to versions before 0.23.0 since only old releases did not depend on DataFrames
TimeModels: recent versions do not depend on DataFrames [add bounds to older releases anyway?]
TimeSeries: recent versions do not depend on DataFrames [add bounds to older releases anyway?]
UAParser: added julia 0.6 upper bound to versions 0.0.2 and 0.0.3 since only these releases did not depend on DataFrames
XSim: already has an upper bound on julia 0.6
OdsIO: added julia 0.6 upper bound to version 0.0.1 since only oldest release did not depend on DataFrames
SpeedDate: added julia 0.6 upper bound to version 0.0.1 since only oldest release did not depend on DataFrames
EasyPhys: added julia 0.6 upper bound to version 0.1.3 since only oldest release did not depend on DataFrames
StatsModels: added julia 0.6 upper bound to versions before 0.1.0 since only oldest releases did not depend on DataFrames
GenomicVectors: only old versions depended on DataFrames, so OK
```

### CODE REVIEW

File: BioMedQuery/versions/0.1.0/requires [hunk 1]
```diff
@@ -1,4 +1,4 @@
-julia 0.4.1
+julia 0.4.1 0.6
```
Comment (tkelman) [inline BioMedQuery/versions/0.1.0/requires]: would be better to add a fake DataFrames dependency here

**Comment by tkelman** [comment]
It's better to add fake DataFrames dependencies than julia upper bounds, unless you're absolutely sure that each version you're adding a julia upper bound to doesn't load at all on that version.

**Comment by nalimilan** [comment]
OK. I've moved to fake DataFrames dependencies for all releases which supported Julia 0.5. For others, there's little chance they would work on 0.6.

**Comment by tkelman** [comment]
Have you verified that? Since DataFrames has been around so long, a fake dependency is nearly harmless.

**Comment by nalimilan** [comment]
TBH I don't really feel like changing all these ~100 releases to use a fake DataFrames dependency. I'd better spend my time on something more productive. Anyway in a few week we'll have 1.0 and people won't even remember 0.6 has existed.

**Comment by tkelman** [comment]
Adding a julia upper bound is pretty drastic and causes errors if anyone has that version installed or pinned, so it should really only be done when it's absolutely known to already not work at all, or there's no other less harmful option. These hundreds of packages are going to prevent anyone from using the new DataFrames for some time with the release having been done in place like this.

**Comment by timholy** [comment]
Possibly useful to anyone dealing with this: `Pkg.checkout("PkgDev", "teh/upperbound_tools")`. https://github.com/JuliaLang/PkgDev.jl/pull/94

**Comment by nalimilan** [comment]
OK, I've changed all Julia upper bounds into fake DataFrames dependencies. Should be good to go now.

> Possibly useful to anyone dealing with this: Pkg.checkout("PkgDev", "teh/upperbound_tools"). JuliaLang/PkgDev.jl#94

@timholy Cool. I wish I had know the existence of this tool before I made the PR... :-/

**Comment by tkelman** [comment]
Note all the comments on that PR, it does not address the issues that made the original PR incomplete. It could automate the text replacement for existing requirements, but doesn't do a completeness check to verify that it will address a compatibility issue without leading to unintended resolution possibilities.

**Comment by timholy** [comment]
Yeah, there's a bit of a conundrum here: it's already useful but making it "industrial strength" is a lot of work. I started it as a charity project but don't intend to finish it unless a desperate personal need arises. I still hold out hope that someone more deeply involved in package management will pick it up someday.

**Comment by aviks** [comment]
@nalimilan shall we merge this and the other one? Looks good to me.

**Comment by tkelman** [comment]
The lower bounds should all be 0.0.0, not 0.0.1

**Comment by nalimilan** [comment]
> The lower bounds should all be 0.0.0, not 0.0.1

I've added a commit to fix this.

**Pull request merged**